### DISCUSSION TITLE
Experimental Pi4 Build

### DISCUSSION THREAD

**Discussion by jnguyen1098** [mrusse/PiMatTimer#4] [Show and tell]
Hey folks, been a big fan of the timer ever since mrusse first designed it. It's a really nice conversation piece that melds together several of the shared interests in our group.

I recently got a 3D printer and had an old RPI 4 lying around, so I wanted to make one of my own despite the build being for PPI 3.

Regarding materials, ignoring the Pi4 and 3D printer, I spent about $60 in extra materials: $30 for the power bank, $20 for the screen, and about $10 in screws, rocker buttons, jumper cable, cherry MX switches, springs (I actually spent more as these came in bulk and I want to be prepared for future projects, but for the sake of this project you won't use much).

I printed this using a Bambu Lab P1S on eSUN's PLA. It was about 100-150g (I don't remember) after tweaking the settings per mrusse's Printable guidelines and my own preferences for stability, so about $4-5 of PLA.

The case provides plenty of room for error and clearance for the wiring: a novice to soldering myself, I was able to get the wires through very easily with tweezers (I will include a list of convenience supplies at the bottom). For the buttons, I ended up using Glorious Pandas, a very tactile switch that is way better than MX Browns in my experience.

The first issue I encountered with the Pi4 is that the ports (ethernet, USB) stick out a few mm more than the Pi3. As a result, the other side of the case did not have enough clearance to fit the entire board. Using a dremel and sawblade on my Swiss Army Knife, I tried to grind down the part that the SD card rests on:
![image](https://github.com/mrusse/PiMatTimer/assets/40174810/77cbc38f-4fe7-473b-960c-b8d81769d7ee)

Likewise, I didn't do a great job and the angling of the board no longer aligned with the tabs that separate the tabs, which I eventually had to remove:
![image](https://github.com/mrusse/PiMatTimer/assets/40174810/f8606de2-6fed-4739-9607-68cd1ae9d75c)

After a bit of wrestling, it did eventually fit. When the assembly was actually done, the mouse cursor kept moving to the top part of the touchscreen, presumably due to the tight fit. I had to deform the PLA near the top of the screen as I assume the tightness caused the frame to register presses where there shouldn't be.

After all of that, and dealing with some Raspbian issues, it currently works.
![image](https://github.com/mrusse/PiMatTimer/assets/40174810/64ab1b53-95e6-4405-9fcf-59459adb2593).

Likewise, the Kilmat made the buttons very thocky and satisfying:
![image](https://github.com/mrusse/PiMatTimer/assets/40174810/6671a4bc-e56d-4307-96e4-f391b96d6420)

That said, I still haven't soldered the rocker button (still debugging some issues on my end) and one of the buttons malfunctioned while assembled (but worked perfectly outside of the case); I may try to make a separate revision for RPI4 but who knows.

Some future ideas:
- Updating the F3d model to support Pi4
- Allowing multiple timers to interface with the webserver for tournaments, races, etc.
- Refactor the code so it doesn't assume the path to be `/home/pi/` as in 2022 Raspbian no longer features default user `pi` so adopters will have to manually create the `pi` user
- More puzzle support
- Design durock or screw-in stabilizers for the buttons so they're more easily pressed at all angles

This is a great project, I enjoyed every second of it, and I like that it has a reasonable skill floor yet a challenging skill ceiling: it's something newcomers like me can work on, yet a veteran could easily augment the design and make it their own (better switches, stabilizing, automating install/timer on startup, supporting other cases/power banks, more access to the GPIO, printing it in resin), and open source projects like this keep me sane.

## Convenience Supplies
- Tweezers - made maneuvering the wires extremely easy
- Heat gun and shrinkwrap for wiring (didn't use this; wish I did)
- Wire strippers - had more difficulty making do without these than anticipated
- Helping hands - make sure you get ones with very rigid arms and a heavy base
- Kilmat/deadening mat - this adds a lot of heft to the buttons and silences any hollow plastic noise, similar to what you'd do to an aluminum keyboard like the Keychron Q1, though increasing the infill and adding more PLA would likely do something similar
- Dialectric grease - until someone comes by with a design that uses stabilizers for the buttons, heavy grease has helped reduce the scratchiness of the button contacting the rest of the case

**Comment by mrusse** [comment]
Great build! Hope you are able to get that button working and get the rocker switch wired up. Originally I thought that the pi 4 would be an almost identical fit, sorry you had to be the pi 4 guinea pig ;)
